test/integration/test_job.py

```diff
@@ -46,6 +46,17 @@ def test_run_program(self, service):
         self.assertEqual(JobStatus.DONE, job.status())
         self.assertTrue(job.result())
 
+    @run_integration_test
+    def test_run_with_simplejson(self, service):
+        """Test retrieving job results with simplejson package installed."""
+        try:
+            __import__("simplejson")
+            job = self._run_program(service=service)
+            job.wait_for_final_state()
+            self.assertTrue(job.result())
+        except ImportError:
+            self.assertRaises(ImportError)
```

Comment:
Have you considered pip installing it, running the test, and pip uninstalling it? Not sure what the best approach (installing it VS raising the import error) is in this case

FWIW, the installation takes very little, so the overall test time should be impacted minimally?

Reply:
I did consider that and yes I think it would be a more complete test that way but I feel that it's generally not good practice to install/uninstall packages inside a test. 

Also not sure what the best approach is here and if there's a better solution

Reply:
I agree that modifying the env to run a test seems not the best approach. But we are doing some similar in the [test](https://github.com/Qiskit/qiskit-ibm-runtime/blob/main/test/integration/test_fake_backends.py#L188) so...I'm also not sure what the best way to test here.

Reply:
Going to merge for now and closely monitor the main branch